react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import { Redirect} from 'react-router-dom';
+import styles from '../../styles.module.scss';
+import Navbar from "../Navbar"
+import { connect } from "react-redux";
+import actionCreators from "../../../../../redux/actions";
+import Spinner from 'react-spinkit';
+
+class Matches extends Component {
+  
+  async componentDidMount(){
+    this.props.dispatch(actionCreators.getMatches());
+  }
+
+  render() {
+    if(!window.localStorage.getItem("token")){
+      return <Redirect to="/"/>;
+    }
+
+    const {matches} = this.props;
+
+    if(this.props.isLoading){
+      return <Spinner name='folding-cube' />;
+    }
+
+    if(this.props.error){
+      return <span>ERROR:  {this.props.error}</span>;
+    }
+
+    return (
```

Comment:
I think all this part should be in a `layout.js` file. This way we separate the smart component with the methods that hold all the logic, from the structure of html. Here's a nice post on the subject: https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0

Reply:
:white_check_mark: